keyboards/gmmk/pro/ansi/keymaps/andrebrait/keymap.c

```diff
@@ -100,8 +149,37 @@ static void start_effects(void);
 #endif
 #define effect_red() r_effect = 0xFF, g_effect = 0x0, b_effect = 0x0
 #define effect_green() r_effect = 0x0, g_effect = 0xFF, b_effect = 0x0
+#define effect_blue() r_effect = 0x0, g_effect = 0x0, b_effect = 0xFF
+#define effect_white() r_effect = 0xFF, g_effect = 0xFF, b_effect = 0xFF
 #endif // RGB_CONFIRMATION_BLINKING_TIME > 0
 
+static uint32_t previous_layer = -1;
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    uint32_t current_layer = get_highest_layer(state);
```

Comment:
```suggestion
    uint8_t current_layer = get_highest_layer(state);
```

uint8_t maxes out at 255, which is still well past what you can have.

Reply:
I know, but the largest datatype for this seems to be uint32_t (and it seems to be what's used for this keyboard), so I decided to just use the same type.

Let me know if I'm wrong.

Reply:
The max value for uint32_t is 4294967295, which is waaay more than you need here.

`get_highest_layer` returns an integer that will be 31 at the most, since layer_state_t is a 32 bit bitmask, at the most.   So, uint32_t is beyond excessive here.  And you should use the smallest type needed/possible, usually.  

so, uint8_t is *more* than sufficient for this. since it supports 0-255.  Which is still well past what you need here. 

It doesn't make a huge different, but it's more a best practice sort of thing.